### ISSUE TITLE
モデル構造の変更について

### ISSUE THREAD

**Issue by burokoron** [thinkout-projects/classify-image#14]
変更するモデル構造の具体的な内容に関して説明します。

現在のモデルの問題点をVGG16を例に説明します。
https://github.com/hiroki-mas-med/classify-image/blob/0b41d2189da4e8d30d7ebba61de7db6a69dcf403/models.py#L194-L216

- 205-206行目のパラメータ数が多く、過学習しやすい
- 207行目は分類クラス数が多くなると精度低下を招く可能性がある
- Keras 1 のAPIを使用している

これらを改善するために以下のモデル構造を提案します。

- 205行目の`Flatten()`の代わりに`GlobalAveragePooling2D()`を使用する
- 206行目の`Dense()`のパラメータ数は直前のレイヤーのパラメータ数の4分の1にする
  - 例えばVGG16だと128になります
- 207行目の`Dropout()`を削除する
- Keras 2 のAPIを使用する
  - [KerasドキュメントのApplications](https://keras.io/ja/applications/)の**Fine-tune InceptionV3 on a new set of classes**を参考にしてください

また、現在、構築するモデル選択は、
https://github.com/hiroki-mas-med/classify-image/blob/0b41d2189da4e8d30d7ebba61de7db6a69dcf403/image_classifier.py#L167-L184
となっていますが、
`model_builder(model='name')`
のような関数を作成することで、コードをすっきりさせませんか？

あと、これは使っていないので削除でいいと思いますがいかがでしょうか？
https://github.com/hiroki-mas-med/classify-image/blob/0b41d2189da4e8d30d7ebba61de7db6a69dcf403/models.py#L218-L245

以上、検討よろしくお願いします。
